app/src/renderer/components/governance/PageProposal.vue

```diff
@@ -183,64 +180,55 @@ export default {
       `votes`,
       `session`
     ]),
-    proposal() {
-      return this.proposals.proposals[this.proposalId]
+    proposal({proposals, proposalId} = this) {
+      return proposals.proposals[proposalId]
     },
-    submittedAgo() {
-      return moment(new Date(this.proposal.submit_time)).fromNow()
+    submittedAgo({proposal} = this) {
+      return moment(new Date(proposal.submit_time)).fromNow()
     },
-    votingStartedAgo() {
-      return moment(new Date(this.proposal.voting_start_time)).fromNow()
+    votingStartedAgo({proposal} = this) {
+      return moment(new Date(proposal.voting_start_time)).fromNow()
     },
-    depositEndsIn() {
-      return moment(new Date(this.proposal.deposit_end_time)).fromNow()
+    depositEndsIn({proposal} = this) {
+      return moment(new Date(proposal.deposit_end_time)).fromNow()
     },
-    totalVotes() {
-      return (
-        this.tally.yes +
-        this.tally.no +
-        this.tally.no_with_veto +
-        this.tally.abstain
-      )
+    totalVotes({tally: {yes, no, no_with_veto, abstain}} = this) {
+      return (yes + no + no_with_veto + abstain)
```

Comment:
Are these guaranteed to be defined? I.e.: `undefined + 1 + undefined = NaN`

Reply:
In theory yes, I tried the counter check and I can't find a way to have this object only partially defined